### ISSUE TITLE
[Proposal] Offer versioned releases

### ISSUE THREAD

**Issue by xelxebar** [PaulBatchelor/Sporth#48]
## Motivation

In the name of making sporth more discoverable and accessible, I would like to package it up for [Void Linux][0]. However, Void upstream does not allow packages compiled from the HEAD of git branches. This is a common policy among the package repositories of other distributions as well.

## Proposal

So, in order to support package maintainers, would it be reasonable to offer [official releases][1]?

[0]:https://voidlinux.org/
[1]:https://help.github.com/en/github/administering-a-repository/creating-releases#automatically-creating-releases

**Comment by PaulBatchelor** [comment]
Thanks, but no thanks.

I am flattered that you want to package Sporth for Void Linux. It seems like an awesome Linux distribution.

However, releases would give people the wrong idea about Sporth.

Firstly, it would imply that I am organized have my act together. I am neither.

Secondly, it would imply that I am actively maintaining and improving Sporth. I am not. While I will continue to accept bugfixes and minor PRs, Sporth development has naturally plateaued. There are reasons for this, but it mainly boils down to Sporth being "good enough", time, and other, better projects that act as successors. Even if I did slap on a release, I wouldn't expect it to change for a while, if ever. And that just feels a bit silly to me.

Thirdly, adding a release tag in this context to me feels like I'd only be doing it to appease the void linux package manager. This idea doesn't sit well with me, as I try not to play favorites between platforms, and also release numbers + tags introduce a new layer of complexity and responsibility, which Sporth isn't ready for (see points 1 and 2).

Sorry about that.

-P

**Comment by xelxebar** [comment]
Oh cool. Thanks for looking at this.

I completely get not wanting to give the wrong impression.. What about using a version like 0.0.x? This is standard practice for "Alpha! Not production ready! Use at your own risk!" software, and simple v0.0.1, v0.0.2, etc. bumps for the rare bugfix doesn't really add "a new layer of complexity."

For software that exists only at patchlevel versions, the expectations for stability and active maintainability are pretty low. If the software is "good enough" for a small community of users, then I say it's "good enough" to let people try packaging it up. The idea itself is pretty darn slick!

Finally, you won't be in danger of playing favorites with Void either. See Arch, Debian, and Gentoo for examples of distribution that relegate software without versioned releases to second class citizens.

Basically, all I'm proposing here is a simple `git tag` and `git push`. At the very least, the idea of Sporth needs to get out there!
